src/services/phone-provider.ts

```diff
@@ -1,38 +1,36 @@
 import twilio from "twilio"
 
 import { baseLogger } from "@services/logger"
+import { getTwilioConfig } from "@config/app"
+import { TwilioError } from "@domain/errors"
 
-const getTwilioClient = () => {
-  const accountSid = process.env.TWILIO_ACCOUNT_SID
-  const apiKey = process.env.TWILIO_API_KEY
-  const apiSecret = process.env.TWILIO_API_SECRET
-
-  const client = twilio(apiKey, apiSecret, { accountSid })
-  return client
-}
+const client = twilio(getTwilioConfig().apiKey, getTwilioConfig().apiSecret, {
+  accountSid: getTwilioConfig().accountSid,
+})
 
 export const sendTwilioText = async ({ body, to, logger }) => {
```

Comment:
These set of abstractions may be a little off here.

The idea here so far has been to define generic interfaces (e.g. `ILightningService`) and generic mehods within then (e.g. `lookupInvoice`) and then to implement these interfaces for specific services (e.g. `lndService` implements the `ILightningService` interface).

If we followed that pattern here, we'd probably more end up with something like an `IPhoneProviderService` interface with generic `sendText` and `getCarrier` methods and defined types, and a `twilioService` that implements this interface.

Reply:
I have refactored the file to make it a class instead of function at the root of the file

Is this not an somewhat non necessary overhead to do an interface here?

Unless I'm mistaken, Typescript will deduce the type from the file anyway. and the interface is really only useful if we plan to have several services implementing the same interface.

Reply:
Its just about separating declaration (which is part of domain) from implementation.
It is useful for testing, clean boundaries, and potentially interchangeability of implementation. We have after all had 2 implementations in the past and now moved back to 1. Its not a lot of overhead and it keeps the structure we have introduced consistent.

Reply:
> Its just about separating declaration (which is part of domain) from implementation.

why should declaration part of `domain` but implementation part of `app`?

Reply:
implementation is part of `services` not `app`. 
Domain is unit-testable and app could be to if we did dependency injection (haven't so far).
> It is useful for testing, clean boundaries, and potentially interchangeability of implementation. We have after all had 2 implementations in the past and now moved back to 1. Its not a lot of overhead and it keeps the structure we have introduced consistent.